scripts/Muon/GUI/Common/fitting_widgets/basic_fitting/basic_fitting_model.py

```diff
@@ -461,6 +511,38 @@ def _get_new_end_x_for(self, new_dataset_name: str) -> float:
             x_lower, x_upper = self.x_limits_of_workspace(new_dataset_name)
             return self.current_end_x if x_lower < self.current_end_x < x_upper else x_upper
 
+    def _get_new_exclude_start_xs_and_end_xs_using_existing_data(self) -> tuple:
+        """Returns the exclude start and end Xs to use for the new data. It tries to use existing ranges if possible."""
+        exclude_start_xs, exclude_end_xs = [], []
+        for dataset_index in range(self.fitting_context.number_of_datasets):
+            exclude_start_x, exclude_end_x = self._get_new_exclude_start_and_end_x_for(dataset_index)
+            exclude_start_xs.append(exclude_start_x)
+            exclude_end_xs.append(exclude_end_x)
+        return exclude_start_xs, exclude_end_xs
+
+    def _get_new_exclude_start_and_end_x_for(self, dataset_index: int) -> tuple:
+        """Gets the new exclude start and end X to use for a specific dataset. It tries to use the current data."""
+        start_x, end_x = self.fitting_context.start_xs[dataset_index], self.fitting_context.end_xs[dataset_index]
+        exclude_start_xs, exclude_end_xs = self.fitting_context.exclude_start_xs, self.fitting_context.exclude_end_xs
+
+        new_exclude_start_x = exclude_start_xs[dataset_index] if dataset_index < len(exclude_start_xs) else end_x
+        new_exclude_end_x = exclude_end_xs[dataset_index] if dataset_index < len(exclude_end_xs) else end_x
+
+        return self._check_new_exclude_start_and_end_x(start_x, end_x, new_exclude_start_x, new_exclude_end_x)
+
+    @staticmethod
+    def _check_new_exclude_start_and_end_x(start_x: float, end_x: float, new_exclude_start_x: float,
```

Comment:
does do the same start and end x checks that we have elsewhere (e.g. for the fitting range)?

Reply:
It's slightly different but I think I'll move it to the `workspace_data_utils`